### ISSUE TITLE
[Feature Request] Option to not suggest links to page that has been already linked to from current file

### ISSUE THREAD

**Issue by kazerniel** [tadashi-aikawa/obsidian-various-complements-plugin#314]
Not sure if this would be difficult to implement, but generally I don't find it useful to link from one page to another multiple times. So it would be quite handy if VC didn't offer link suggestions to pages that have been already linked from the current page. I most often run into this in my journals, when as I'm talking about a topic, VC keeps and keeps suggesting to link each time I mention the page name, when I already linked to it earlier.

I guess there could be performance concerns for very large documents, but maybe it could be implemented with a slider similar to other existing settings.

**Comment by diiiv9** [comment]
@kazerniel I had a similar  [issue](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/issues/240) , I felt I still needed to insert the completion but not as a link , so the option to _insert as text_ was added. Maybe this could help ?

**Comment by kazerniel** [comment]
Thanks @diiiv9! Ideally I'd like an option to not get a suggestion at all in this scenario (as I only use link complements), but until such a feature could be added, I'll try to remember to use your solution.

**Comment by tadashi-aikawa** [comment]
Hi, @kazerniel :)

> VC keeps and keeps suggesting to link each time I mention the page name, when I already linked to it earlier.

I’ve also often been concerned about this issue. However, I hesitated to add the feature because it goes hand in hand with the problem that if we fix the link to a single note, the relevant information could disappear when that sentence is deleted.

That said, such cases are probably rare, and I agree that there wouldn't be any major concerns if we make it an option. I'll consider addressing it 👍

**Comment by georgeamccarthy** [comment]
I would also find this feature very useful. I try to create Obsidian note links like Wikipedia article links, only creating them for the first instance of a concept on the page.

**Comment by BrandonLMcintosh** [comment]
Agreed. I take notes on technological concepts, and having the option not to repeat links multiple times in the same file would have changed my entire vault for the better. 

+1 to add an option for internal link compliment limits.

**Comment by tadashi-aikawa** [comment]
@kazerniel @diiiv9 @georgeamccarthy @BrandonLMcintosh 
Released in [v10.4.0](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/10.4.0) 🚀
